src/cpu/aarch64/jit_sve_512_x8s8s32x_conv_kernel.cpp

```diff
@@ -0,0 +1,1642 @@
+                int penultimate_iw_leeway = (jcp.iw - 1 - penultimate_iw_index)
+                                * jcp.ic_without_padding
+                        + jcp.ic_without_padding % broadcast_size;
+                if (penultimate_iw_leeway >= broadcast_size) break;
+                --jcp.ur_w;
+            }
+            if (jcp.ur_w == 0) // no satisfactory ur_w could be found
+                return status::unimplemented;
+        }
+    }
+    jcp.ur_w_tail = jcp.ow % jcp.ur_w;
+
+    jcp.ow_block = jcp.ow;
+    int base_work_amount = jcp.mb * jcp.nb_ch * jcp.od * jcp.oh
+            * (jcp.nb_oc / jcp.nb_oc_blocking_thr_chunk);
+    float best_thr_eff
+            = (float)base_work_amount / rnd_up(base_work_amount, jcp.nthr);
+    int max_nb_ow = div_up(jcp.ow, 2 * jcp.ur_w);
+    for (int nb_ow = 1; nb_ow <= max_nb_ow; nb_ow++) {
+        int ow_block
+                = nstl::min(rnd_up(div_up(jcp.ow, nb_ow), jcp.ur_w), jcp.ow);
+        if (ow_block < jcp.nb_oc_blocking_thr_chunk * jcp.oc_block
+                && best_thr_eff > 0.8f)
+            break;
+        if (div_up(jcp.ow, ow_block) != nb_ow) continue;
+        auto work_amount = base_work_amount * nb_ow;
+        float thr_eff = (float)work_amount / rnd_up(work_amount, jcp.nthr);
+        if (ow_block >= 2 * jcp.ur_w && thr_eff > 1.1f * best_thr_eff) {
+            jcp.ow_block = ow_block;
+            best_thr_eff = thr_eff;
+        }
+        if (best_thr_eff > 0.9f) break;
+    }
+    jcp.nb_ow = div_up(jcp.ow, jcp.ow_block);
+
+    bool args_ok = true && jcp.oc % jcp.oc_block == 0 && jcp.l_pad <= jcp.ur_w;
+    if (!args_ok) return status::unimplemented;
+
+    int r_pad_no_tail = nstl::max(0,
+            calculate_end_padding(jcp.l_pad, jcp.ow - jcp.ur_w_tail, jcp.iw,
+                    jcp.stride_w, ext_kw));
+    if (r_pad_no_tail > jcp.ur_w) return status::unimplemented;
+
+    pick_loop_order(jcp, jcp.nthr);
+
+    jcp.nb_ic_L2 = jcp.nb_ic;
+
+    const auto &oscales = attr.output_scales_;
+    jcp.is_oc_scale = oscales.mask_ == 1 << 1;
+
+    // only common and per-oc-channel scales are supported
+    const bool oscales_ok = one_of(oscales.mask_, 0, 1 << 1);
+    if (!oscales_ok) return status::unimplemented;
+
+    jcp.wei_adj_scale
+            = (weights_d.extra().flags & memory_extra_flags::scale_adjust)
+            ? weights_d.extra().scale_adjust
+            : 1.f;
+
+    int ch_block = jcp.is_depthwise ? jcp.ch_block : jcp.ic_block;
+    if (ch_block != 16) return status::unimplemented;
+
+    return status::success;
+}
+
+void jit_sve_512_x8s8s32x_fwd_kernel::init_scratchpad(
+        memory_tracking::registrar_t &scratchpad, const jit_conv_conf_t &jcp,
+        const primitive_attr_t &attr) {}
+
+template struct _jit_sve_512_x8s8s32x_fwd_kernel<ZReg>;
+#if 0
+template struct _jit_sve_512_x8s8s32x_fwd_kernel<Ymm>;
+template struct _jit_sve_512_x8s8s32x_fwd_kernel<Xmm>;
+#endif
```

Comment:
^^ Above [Lines 1636--1638 and 1625] 
we don't deal with  `channel_block = 4 or 8`  scenario. But rest of the code seems generic enough [especially with regards to weights tag, etc.] to handle it as well. So I was wondering why it is being left out

Reply:
Thx! I'll delete it.

Reply:
We are preparing sve_512 source codes that deal with `channel_block = 16, 8, and 4` scenario. We found the following definitions `template struct` unnecessary. Therefore, we will delete them.

```
template struct _jit_sve_512_x8s8s32x_fwd_kernel<ZReg>;
#if 0
template struct _jit_sve_512_x8s8s32x_fwd_kernel<Ymm>;
template struct _jit_sve_512_x8s8s32x_fwd_kernel<Xmm>;
#endif
```